File: 2114-maximum-number-of-words-found-in-sentences/2114-maximum-number-of-words-found-in-sentences.java
class Solution {
    public int mostWordsFound(String[] sentences) {
        //logic count space in each senetences
        //return count+1
        
        int res=0;
        int currentCount=0;
        for(String s:sentences){
            for(int i=0;i<s.length();i++){
                if(s.charAt(i)==' '){
                    currentCount+=1;
                }
            }
            res=Math.max(currentCount,res);
            currentCount=0;
        }
        return res+1;
    }
}

File: 2114-maximum-number-of-words-found-in-sentences/2114-maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        
        int currentCount;
        int res=0;
        for(int i=0;i<sentences.size();i++){
            currentCount=1;
            for(int j=0;j<sentences[i].size();j++){
                if(sentences[i][j]==' '){
                    currentCount++;
                }
            }
            res= max(currentCount,res);
        }
        return res;
    }
};